From 95a8976c7d0d55a88caec62f5f5d37ac69806f03 Mon Sep 17 00:00:00 2001
Message-ID: <95a8976c7d0d55a88caec62f5f5d37ac69806f03.1708115863.git.mjg@fedoraproject.org>
In-Reply-To: <1932a672db047da3204a445880007fcc522fa7d7.1708115863.git.mjg@fedoraproject.org>
References: <1932a672db047da3204a445880007fcc522fa7d7.1708115863.git.mjg@fedoraproject.org>
From: Julian Smith <julian.smith@artifex.com>
Date: Thu, 14 Sep 2023 18:57:42 +0100
Subject: [PATCH 04/11] Makerules: fixes for shared shared library installs on
 OpenBSD.

We need to use the same pkg-config calls as Linux.

(cherry picked from commit fbf12be21348aa2ea8977a61d1dac1fcf04c36ed)
---
 Makerules | 23 +++++++++++++++++++----
 1 file changed, 19 insertions(+), 4 deletions(-)

diff --git a/Makerules b/Makerules
index 916e8581a..49c04f009 100644
--- a/Makerules
+++ b/Makerules
@@ -162,6 +162,13 @@ ifneq "$(CLUSTER)" ""
   CFLAGS += -DCLUSTER
 endif
 
+ifeq ($(OS),Linux)
+    LINUX_OR_OPENBSD := yes
+endif
+ifeq ($(OS),OpenBSD)
+    LINUX_OR_OPENBSD := yes
+endif
+
 ifeq ($(OS),MINGW)
   WINDRES := windres
   HAVE_WIN32 := yes
@@ -188,8 +195,11 @@ else ifeq ($(OS),MACOS)
     endif
   endif
 
-else ifeq ($(OS),Linux)
-  HAVE_OBJCOPY := yes
+else ifeq ($(LINUX_OR_OPENBSD),yes)
+
+  ifeq ($(OS),Linux)
+    HAVE_OBJCOPY := yes
+  endif
 
   ifeq ($(shell pkg-config --exists freetype2 && echo yes),yes)
 	SYS_FREETYPE_CFLAGS := $(shell pkg-config --cflags freetype2)
@@ -240,8 +250,13 @@ else ifeq ($(OS),Linux)
 
   HAVE_GLUT := yes
   ifeq ($(HAVE_GLUT),yes)
-	SYS_GLUT_CFLAGS :=
-	SYS_GLUT_LIBS := -lglut -lGL
+    ifeq ($(OS),OpenBSD)
+      SYS_GLUT_CFLAGS := $(shell pkg-config --cflags glut gl)
+      SYS_GLUT_LIBS := $(shell pkg-config --libs glut gl)
+    else
+      SYS_GLUT_CFLAGS :=
+      SYS_GLUT_LIBS := -lglut -lGL
+    endif
   endif
 
   HAVE_X11 := $(shell pkg-config --exists x11 xext && echo yes)
-- 
2.44.0.rc1.222.g52f20dec8d

